docs/component-guides/profiles-irsa.md

```diff
@@ -0,0 +1,118 @@
+# IAM Role Service Accounts (IRSA) support for Kubeflow Profiles
+
+# Background
+
+In AWS, [IRSA](https://aws.amazon.com/blogs/opensource/introducing-fine-grained-iam-roles-service-accounts/) is a permissions management tool that allows applying AWS IAM permissions boundaries to Kubernetes service accounts (SA). This is useful, for example, to grant one SA permissions to upload test results to a certain bucket in S3 while granting another SA permissions to read from that bucket without modifying the contents. Because the IAM permissions apply to the SA, the permissions boundary is limited to the pod level rather than the nodegroup level. 
+
+In Kubeflow, IRSA is used to provide an isolation boundary at the profile level, allowing admins to scope profiles to their necessary IAM permissions to meet their AWS usage requirements. This is done by making the profile SA an IRSA and by creating profiles with the `AwsIamForServiceAccount` plugin.
+
+Below are the steps to configure IRSA to be used with Kubeflow Profiles.
+
+# Configuration Steps
+
+The following steps should be run after deploying Kubeflow via the [provided deployment options](../../docs/deployment)
+
+1. Export the following variables for convenience
+    ```
+    export CLUSTER_NAME=<your cluster name>
+    export CLUSTER_REGION=<your region>
+    export AWS_ACCOUNT_ID=$(aws sts get-caller-identity --query "Account" --output text)
+    export PROFILE_NAME=<the name of the profile to be created>
+    ```
+
+1. Create an IAM policy using [the following policy document](../../awsconfigs/infra_configs/iam_profile_controller_policy.json)
+    ```
+    aws iam create-policy \
+    --region $CLUSTER_REGION \
+    --policy-name kf-profile-controller-policy \
+    --policy-document file://awsconfigs/infra_configs/iam_profile_controller_policy.json
+    ```
+
+1. Associate IAM OIDC with the cluster
+    ```
+    aws --region $CLUSTER_REGION eks update-kubeconfig --name $CLUSTER_NAME
+
+    eksctl utils associate-iam-oidc-provider --cluster $CLUSTER_NAME --region $CLUSTER_REGION --approve
+    ```
+
+1. Create an IRSA for the profile controller using the policy.
+    ```
+    eksctl create iamserviceaccount \
+    --cluster=$CLUSTER_NAME \
+    --name="profiles-controller-service-account" \
+    --namespace=kubeflow \
+    --attach-policy-arn="arn:aws:iam::${AWS_ACCOUNT_ID}:policy/kf-profile-controller-policy" \
+    --region=$CLUSTER_REGION \
+    --override-existing-serviceaccounts \
+    --approve
+    ```
+
+1. Create an IAM trust policy to authorize federated requests from the OIDC provider
+    ```
```

Comment:
FYI: this indentation for the code block does not render in the README, it is helpful in keeping the numbering? if not maybe add a bullet - so the indent is rendered?

Reply:
numbering seems to be preserved: https://github.com/rrrkharse/kubeflow-manifests/tree/profiles-irsa/docs/deployment/add-ons/profile-iam

don't seem to need the indentation right now, can add it later if needed if the steps get more complex